api/index.ts

```diff
@@ -0,0 +1,113 @@
+import axios from 'axios';
+import mitt from 'mitt'
+import {
+  StatusCodes
+} from 'http-status-codes';
+import { setupCache } from 'axios-cache-adapter'
+import { events } from '../types';
+
+const emitter = mitt();
+
+let token = ''
+let instanceUrl = ''
+let cacheMaxAge = 0
+
+export function updateToken(key: string) {
+  token = key
+}
+
+export function updateInstanceUrl(url: string) {
+  instanceUrl = url
+}
+
+export function init(key: string, url: string, cacheAge: number) {
+  token = key
+  instanceUrl = url
+  cacheMaxAge = cacheAge
+}
+
+axios.interceptors.request.use(async (config: any) => {
+    if (token) {
+        config.headers.Authorization =  'Bearer ' + token;
+        config.headers['Content-Type'] = 'application/json';
+    }
+    return config;
+});
+
+axios.interceptors.response.use(function (response) {
+    // Any status code that lie within the range of 2xx cause this function to trigger
+    // Do something with response data
+    return response;
+  }, function (error) {
+    // TODO Handle it in a better way
+    // Currently when the app gets offline, the time between adding a loader and removing it is fractional due to which loader dismiss is called before loader present
+    // which cause loader to run indefinitely
+    // Following gives dismiss loader a delay of 100 microseconds to get both the actions in sync
+    setTimeout(() => emitter.emit("dismissLoader"), 100);
```

Comment:
We should include 'dismissLoader' to event types

Reply:
Done Sir